llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp

```diff
@@ -1391,9 +1452,35 @@ bool SPIRVInstructionSelector::selectFCmp(Register ResVReg,
 
 Register SPIRVInstructionSelector::buildZerosVal(const SPIRVType *ResType,
                                                  MachineInstr &I) const {
+  // OpenCL uses nulls for Zero. In HLSL we don't use null constants.
+  bool ZeroAsNull = STI.isOpenCLEnv();
+  if (ResType->getOpcode() == SPIRV::OpTypeVector)
+    return GR.getOrCreateConstVector(0UL, I, ResType, TII, ZeroAsNull);
+  return GR.getOrCreateConstInt(0, I, ResType, TII, ZeroAsNull);
+}
+
+APFloat getZeroFP(const Type *LLVMFloatTy) {
```

Comment:
Could we make the function `static`?

Reply:
Sure. I was the only user so wasn't sure if it made sense to make it into a static class function.

Reply:
I mean just a free standing static fun, like isGenericCastablePtr or isUSMStorageClass